.github/workflows/build.yml

```diff
@@ -0,0 +1,40 @@
+name: .NET Library
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ windows-latest ] # Add other OSes later
+        dotnet: ['5.0.x']
+
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+
+      - name: Install Codecov tool
+        run: dotnet tool install --global Codecov.Tool --version 1.13.0
+
+      - name: Build, unit test & generate coverage report
+        run: dotnet test --configuration Release --verbosity normal /p:CollectCoverage=true /p:CoverletOutputFormat=opencover
```

Comment:
I see this test command looks different from the command that was in `test.yml` - there are options like --no-build, etc. Should they be carried over?

Reply:
The previous GitHub Action wasn't even doing anything because it wasn't building the project or installing the dependencies.

no-build is for if you already have tried compiling the project as a separate step (which we don't do)

no-restore is if we already downloaded the project dependencies for the build... which we also never did